Platformsh.php

```diff
@@ -457,4 +458,48 @@ protected function generatePassword($password)
             ]
         );
     }
+
+    /**
+     * Execute compilation command.
+     * Doing multi-tenant compilation always, because there is known issue with single-tenant one
+     */
+    protected function doCompilation()
+    {
+        $this->log("Removing old compilation files");
+        $this->execute('rm -rf var/di/*');
+        $this->log("Doing code compilation");
+        $this->execute(
+            "cd bin/; /usr/bin/php ./magento setup:di:compile-multi-tenant"
+        );
+    }
+
+    /**
+     * If current deploy is about master branch
+     * 
+     * @return boolean
+     */
+    protected function isProductionMode()
+    {
+        if (is_null($this->isProductionMode)) {
+            if (isset($_ENV["PLATFORM_ENVIRONMENT"]) && $_ENV["PLATFORM_ENVIRONMENT"] == self::GIT_MASTER_BRANCH) {
+                $this->isProductionMode = true;
+            } else {
+                $this->isProductionMode = false;
+            }
+        }
+        return $this->isProductionMode;
+    }
```

Comment:
I don't think this is desirable. All environments on Platform.sh should use the same configuration as production, or you are not able to meet the promise of "if it works in an staging, it works in production".

Platform.sh doesn't have the concept of a "development" environment that would have a different configuration (of DI compilation, of asset minimizing, etc.).